docs/reference/content/reference/unified-topology/index.md

```diff
@@ -0,0 +1,78 @@
++++
+date = "2019-07-04T12:53:30-04:00"
+title = "Unified Topology Design"
+[menu.main]
+  parent = "Reference"
+  identifier = "Unified Topology Design"
+  weight = 100
+  pre = "<i class='fa'></i>"
++++
+
+# Unified Topology Design
+
+At the time of writing the node driver has seven topology classes, including the newly introduced unified topology. Each legacy topology type from the core module targets a supported topology class: Replica Sets, Sharded Deployments (mongos) and Standalone servers. On top of each of these rests a thin topology wrapper from the "native" layer which introduces the concept of a "disconnect handler", essentially a callback queue for handling naive retryability.
+
+The goal of the unified topology is threefold:
+  - fully support the drivers Server Discovery and Monitoring, Server Selection and Max Staleness specifications
+  - reduce the maintenance burden of supporting the topology layer in the driver by modeling all supported topology types with a single engine
+  - remove confusing functionality which could be potentially dangerous for our users
+
+## How to use it
+
+The unified topology is available now behind the `useUnifiedTopology` feature flag. You can opt in to using it by passing the option to your `MongoClient` constructor:
+
+```js
+const client = MongoClient('mongodb://localhost:27017', { useUnifiedTopology: true });
+```
+
+## Behavioral Changes
+
+### `MongoClient.connect`, `isConnected`
+
+The unified topology is the first step in a paradigm shift away from a concept of "connecting" to a MongoDB deployment using a `connect` method. Consider for a moment what it means to be connected to a replica set: do we trigger this state when connected to a primary? A primary and one secondary? When connected to all known nodes? It's unclear whether its possible to answer this without introducing something like a `ReadPreference` parameter to the `connect` method. At this point "connecting" is just one half of "operation execution" - you pass a `ReadPreference` in, and await a selectable server for the operation, now we're connected!
+
+But couldn't you do all of that as a part of your first operation? Our goal is to move towards code that looks more like the following:
+
+```js
+const client = new MongoClient('mongodb://llama:drama@localhost:27017/?replicaSet=rs');
+const coll = client.db('test').collection('foo');
+await coll.insert({ test: 'document' });
+const docs = coll.find({ test: 1 }, { readPreference: 'secondary' }).toArray();
+console.dir({ docs });
+await client.close();
+```
+
+A default `ReadPreference` of "primary" is used for the first write, and a part of awaiting that insert involves initiating connections to all servers in a cluster, selecting a server and executing the operation. Errors will surface at the callsite of any given operation, giving the user more fine-grained control over error handling.
+
+#### Why does `MongoClient.isConnected` always return `true`?
```

Comment:
Probably add a note that `isConnected` will be superfluous soon / go away?

Reply:
added a note for this